File: src/main/scala/havarunner/TestAndParameters.scala
package havarunner

import org.junit.runners.model.{TestClass, FrameworkMethod}
import java.lang.reflect.Method

private[havarunner] class TestAndParameters(
  val frameworkMethod: FrameworkMethod,
  val testClass: TestClass,
  val scenario: Object,
  val beforeClasses: Seq[Method],
  val befores: Seq[Method],
  val afters: Seq[Method],
  val afterClasses: Seq[Method],
  val runSequentially: Boolean
) {
  def scenarioToString = {
    if (scenario == ScenarioHelper.defaultScenario) {
      ""
    } else {
      s" (when ${scenario.toString})"
    }
  }
}

File: src/main/java/havarunner/TestAndParameters.java
package havarunner;

import org.junit.runners.model.FrameworkMethod;
import org.junit.runners.model.TestClass;

import java.lang.reflect.Method;
import java.util.Collection;

class TestAndParameters {
    final FrameworkMethod frameworkMethod;
    final TestClass testClass;
    final Object scenario;
    final Collection<Method> befores;

    TestAndParameters(
        FrameworkMethod frameworkMethod,
        TestClass testClass,
        Object scenario,
        Collection<Method> befores
    ) {
        this.frameworkMethod = frameworkMethod;
        this.testClass = testClass;
        this.scenario = scenario;
        this.befores = befores;
    }

     String scenarioToString() {
        if (scenario.equals(ScenarioHelper.defaultScenario)) {
            return "";
        } else {
            return " (when " + scenario.toString() + ")";
        }
    }
}
